**Reading Data**

In [1]:
import pandas as pd
import numpy as np 

autoinsurance_churn =pd.read_csv("/home/jampanasasank/Desktop/Customer Churn Prediction/Customer-Churn-Prediction/AllData/archive (6)/autoinsurance_churn.csv")

In [2]:
autoinsurance_churn.shape

(1680909, 22)

In [3]:
autoinsurance_churn.head()

individual_id    address_id  curr_ann_amt  days_tenure cust_orig_date  \
0   2.213000e+11  5.213000e+11    818.877997       1454.0     2018-12-09   
1   2.213001e+11  5.213001e+11    974.199182       1795.0     2018-01-02   
2   2.213007e+11  5.213002e+11    967.375112       4818.0     2009-09-23   
3   2.213016e+11  5.213006e+11    992.409561        130.0     2022-07-25   
4   2.213016e+11  5.213006e+11    784.633494       5896.0     2006-10-11   

   age_in_years date_of_birth   latitude  longitude           city  ...  \
0            44    1978-06-23  32.578829 -96.305006        Kaufman  ...   
1            72    1950-05-30  32.732209 -97.000893  Grand Prairie  ...   
2            55    1967-07-07  32.819777 -96.846938         Dallas  ...   
3            53    1969-05-25  32.684065 -97.162180      Arlington  ...   
4            50    1972-09-25  32.751398 -97.376745     Fort Worth  ...   

     income has_children  length_of_residence  marital_status  \
0   22500.0          1.0                 15.0         Married   
1   27500.0          0.0                  2.0          Single   
2   42500.0          0.0                 10.0         Married   
3  125000.0          1.0                  6.0         Married   
4   87500.0          1.0                  4.0         Married   

   home_market_value home_owner college_degree  good_credit  acct_suspd_date  \
0      50000 - 74999        1.0            1.0          1.0              NaN   
1      50000 - 74999        1.0            0.0          0.0              NaN   
2      75000 - 99999        1.0            0.0          0.0              NaN   
3    175000 - 199999        1.0            0.0          1.0       2021-12-22   
4    225000 - 249999        1.0            1.0          1.0              NaN   

   Churn  
0      0  
1      0  
2      0  
3      1  
4      0  

[5 rows x 22 columns]

In [4]:
autoinsurance_churn.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1680909 entries, 0 to 1680908
Data columns (total 22 columns):
 #   Column               Non-Null Count    Dtype  
---  ------               --------------    -----  
 0   individual_id        1680909 non-null  float64
 1   address_id           1680909 non-null  float64
 2   curr_ann_amt         1680909 non-null  float64
 3   days_tenure          1680909 non-null  float64
 4   cust_orig_date       1680909 non-null  object 
 5   age_in_years         1680909 non-null  int64  
 6   date_of_birth        1680909 non-null  object 
 7   latitude             1427190 non-null  float64
 8   longitude            1427190 non-null  float64
 9   city                 1668842 non-null  object 
 10  state                1680909 non-null  object 
 11  county               1668842 non-null  object 
 12  income               1680909 non-null  float64
 13  has_children         1680909 non-null  float64
 14  length_of_residence  1680909 non-null  float64
 15

Data have 22 columns in total where 21 columns are independent and one dependent column (Target column). 

it has 14 numerical columns and 8 categorical columns.

In [5]:
autoinsurance_churn.nunique()

individual_id          1680909
address_id             1296697
curr_ann_amt           1679315
days_tenure               5244
cust_orig_date            5244
age_in_years                92
date_of_birth             1085
latitude                715140
longitude               721161
city                        95
state                        1
county                      14
income                      20
has_children                 2
length_of_residence         17
marital_status               2
home_market_value           19
home_owner                   2
college_degree               2
good_credit                  2
acct_suspd_date            300
Churn                        2
dtype: int64

**individual id is the unique key here**

In [6]:
autoinsurance_churn.drop_duplicates(subset=['individual_id'], inplace=True)

In [7]:
autoinsurance_churn.shape

(1680909, 22)

**Target column**

In [8]:
autoinsurance_churn['Churn'].value_counts()

Churn
0    1487453
1     193456
Name: count, dtype: int64

In [9]:
float(autoinsurance_churn['Churn'].value_counts()[0]/autoinsurance_churn['Churn'].shape[0])*100

88.49098910173008

In [10]:
float(autoinsurance_churn['Churn'].value_counts()[1]/autoinsurance_churn['Churn'].shape[0])*100

11.509010898269924

We can observe a clear imbalance here, Customer's who are not churning is 88.5%, which means around 90% of the total customers aren't churning and only 10% are likely to leave.

In [11]:
# from dataprep.eda import create_report
# report = create_report(autoinsurance_churn, title='Autoinsurance_Churn_Report')

In [12]:
data = autoinsurance_churn.copy()

In [13]:
data

individual_id    address_id  curr_ann_amt  days_tenure  \
0         2.213000e+11  5.213000e+11    818.877997       1454.0   
1         2.213001e+11  5.213001e+11    974.199182       1795.0   
2         2.213007e+11  5.213002e+11    967.375112       4818.0   
3         2.213016e+11  5.213006e+11    992.409561        130.0   
4         2.213016e+11  5.213006e+11    784.633494       5896.0   
...                ...           ...           ...          ...   
1680904   2.213007e+11  5.213002e+11   1259.900413        803.0   
1680905   2.213015e+11  5.213005e+11    604.096865       3261.0   
1680906   2.213026e+11  5.213009e+11   1255.570597       3403.0   
1680907   2.213008e+11  5.213003e+11   1104.105051       1258.0   
1680908   2.213016e+11  5.213006e+11    829.869558       2504.0   

        cust_orig_date  age_in_years date_of_birth   latitude  longitude  \
0           2018-12-09            44    1978-06-23  32.578829 -96.305006   
1           2018-01-02            72    1950-05-30  32.732209 -97.000893   
2           2009-09-23            55    1967-07-07  32.819777 -96.846938   
3           2022-07-25            53    1969-05-25  32.684065 -97.162180   
4           2006-10-11            50    1972-09-25  32.751398 -97.376745   
...                ...           ...           ...        ...        ...   
1680904     2020-09-20            55    1967-07-07  32.678483 -96.665119   
1680905     2013-12-28            77    1945-07-01  32.972007 -96.688905   
1680906     2013-08-08            41    1981-09-22  32.902815 -96.510684   
1680907     2019-06-23            52    1970-07-26  32.892940 -97.245317   
1680908     2016-01-24            64    1959-04-29  33.146851 -96.968217   

                  city  ...    income has_children  length_of_residence  \
0              Kaufman  ...   22500.0          1.0                 15.0   
1        Grand Prairie  ...   27500.0          0.0                  2.0   
2               Dallas  ...   42500.0          0.0                 10.0   
3            Arlington  ...  125000.0          1.0                  6.0   
4           Fort Worth  ...   87500.0          1.0                  4.0   
...                ...  ...       ...          ...                  ...   
1680904         Dallas  ...   27500.0          1.0                  3.0   
1680905     Richardson  ...   87500.0          0.0                  1.0   
1680906        Rowlett  ...  125000.0          0.0                 12.0   
1680907         Keller  ...   87500.0          1.0                 11.0   
1680908     Little Elm  ...  225000.0          1.0                  4.0   

         marital_status  home_market_value home_owner college_degree  \
0               Married      50000 - 74999        1.0            1.0   
1                Single      50000 - 74999        1.0            0.0   
2               Married      75000 - 99999        1.0            0.0   
3               Married    175000 - 199999        1.0            0.0   
4               Married    225000 - 249999        1.0            1.0   
...                 ...                ...        ...            ...   
1680904          Single      50000 - 74999        1.0            0.0   
1680905         Married    125000 - 149999        1.0            0.0   
1680906         Married    125000 - 149999        1.0            1.0   
1680907          Single      75000 - 99999        1.0            1.0   
1680908         Married    200000 - 224999        1.0            0.0   

         good_credit  acct_suspd_date  Churn  
0                1.0              NaN      0  
1                0.0              NaN      0  
2                0.0              NaN      0  
3                1.0       2021-12-22      1  
4                1.0              NaN      0  
...              ...              ...    ...  
1680904          1.0              NaN      0  
1680905          1.0              NaN      0  
1680906          1.0              NaN      0  
1680907          1.0              NaN      0  
1680908 

In [15]:
data.iloc[0]

individual_id          221300017872.0
address_id             521300005982.0
curr_ann_amt               818.877997
days_tenure                    1454.0
cust_orig_date             2018-12-09
age_in_years                       44
date_of_birth              1978-06-23
latitude                    32.578829
longitude                  -96.305006
city                          Kaufman
state                              TX
county                        Kaufman
income                        22500.0
has_children                      1.0
length_of_residence              15.0
marital_status                Married
home_market_value       50000 - 74999
home_owner                        1.0
college_degree                    1.0
good_credit                       1.0
acct_suspd_date                   NaN
Churn                               0
Name: 0, dtype: object

In [ ]:
#cust_orig_date, date_of_birth, acc_suspd_date, home_market_value

In [27]:
np.mean([int(i) for i in data['home_market_value'][0].split('-')])

np.float64(62499.5)

In [26]:
np.mean([10,20])

np.float64(15.0)